testing/src/test/java/io/stargate/it/cql/BoundStatementTest.java

```diff
@@ -14,42 +15,51 @@
 import com.datastax.oss.driver.api.core.data.ByteUtils;
 import com.datastax.oss.driver.api.core.servererrors.InvalidQueryException;
 import com.datastax.oss.driver.api.core.servererrors.ProtocolError;
+import io.stargate.it.BaseOsgiIntegrationTest;
+import io.stargate.it.driver.CqlSessionExtension;
+import io.stargate.it.driver.CqlSessionSpec;
+import io.stargate.it.storage.StargateConnectionInfo;
+import io.stargate.it.storage.StargateEnvironmentInfo;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
 /**
  * Note: all tests related to unset values assume that the protocol version is always 4 or higher.
  */
-public class BoundStatementTest extends JavaDriverTestBase {
+@ExtendWith(CqlSessionExtension.class)
+@CqlSessionSpec(
+    initQueries = {
+      // table where every column forms the primary key.
+      "CREATE TABLE IF NOT EXISTS test (k text, v int, PRIMARY KEY(k, v))",
+      // table with simple primary key, single cell.
+      "CREATE TABLE IF NOT EXISTS test2 (k text primary key, v0 int)",
+      // table with composite partition key
+      "CREATE TABLE IF NOT EXISTS test3 (pk1 int, pk2 int, v int, PRIMARY KEY ((pk1, pk2)))",
+    })
+public class BoundStatementTest extends BaseOsgiIntegrationTest {
 
   private static final String KEY = "test";
   private static final int VALUE = 7;
 
   @BeforeEach
-  public void setupSchema() {
-    // table where every column forms the primary key.
-    session.execute("CREATE TABLE IF NOT EXISTS test (k text, v int, PRIMARY KEY(k, v))");
+  public void cleanupData(CqlSession session) {
+    session.execute("TRUNCATE test");
+    session.execute("TRUNCATE test2");
+    session.execute("TRUNCATE test3");
+
     for (int i = 0; i < 100; i++) {
       session.execute(
           SimpleStatement.builder("INSERT INTO test (k, v) VALUES (?, ?)")
               .addPositionalValues(KEY, i)
               .build());
     }
-
-    // table with simple primary key, single cell.
-    session.execute("CREATE TABLE IF NOT EXISTS test2 (k text primary key, v0 int)");
-
-    // table with composite partition key
-    session.execute(
-        "CREATE TABLE IF NOT EXISTS test3 "
-            + "(pk1 int, pk2 int, v int, "
-            + "PRIMARY KEY ((pk1, pk2)))");
   }
 
   @Test
   @DisplayName("Should execute statement with positional values")
-  public void positionalValuesTest() {
+  public void positionalValuesTest(CqlSession session) {
```

Comment:
can't we inject the `CqlSession session` in the test constructor and use it in methods instead of injecting for every test method?
(similar to what we are doing in java-driver tests)

Reply:
Constructor parameter injection should work technically. Still, it worth with `@ParameterizedTest` as well.

As for me, I kind of like having per-method parameters, personally :)

Reply:
I think either approach will work when using a class-level annotation. Not so sure about method-level though, I don't think JUnit creates a new test class instance for every method.

I like passing the session as a parameter because it lifts any ambiguity. It's not so bad with parameterized tests, you can put it last.